app/util/sentry/utils.js

```
@@ -117,6 +350,47 @@ function removeSES(report) {
   }
 }
 
+/**
+ * Return a "masked" copy of the given object. The returned object includes
+ * only the properties present in the mask.
+ *
+ * The mask is an object that mirrors the structure of the given object, except
+ * the only values are `true`, `false, a sub-mask, or the 'AllProperties"
+ * symbol. `true` implies the property should be included, and `false` will
+ * exclude it. A sub-mask implies the property should be further masked
+ * according to that sub-mask. The "AllProperties" symbol is used for objects
+ * with dynamic keys, and applies a rule (either `true`, `false`, or a
+ * sub-mask`) to every property in that object.
+ *
+ * If a property is excluded, its type is included instead.
+ *
+ * @param {object} object - The object to mask
+ * @param {{[key: string]: object | boolean}} mask - The mask to apply to the object
+ */
+export function maskObject(object, mask) {
+  if (!object) return {};
```

S: could you test the mask param too? Or maybe have a default value for this function mask param?

---

Addressed on this commit: https://github.com/MetaMask/metamask-mobile/pull/10241/commits/26ba148811d8c108dfe5ebb850e4a5fe17d9a232